### PR TITLE
Count bids with function on adapter if present

### PR DESCRIPTION
**Pull request by protonate**
## Type of change
- [x] Other

## Description of change
We count bids expected so we know when to invoke callbacks for a placement and for the overall request. To support counting bids in different ways this PR will use a function `countBids`, if available, in place of the default behavior.

### CODE REVIEW

File: src/adapters/indexExchange.js [hunk 1]
```diff
@@ -530,8 +530,14 @@ var IndexExchangeAdapter = function IndexExchangeAdapter() {
...
 
   return {
-    callBids: _callBids
+    callBids: _callBids,
+    countBids: _countBids
   };
 };
 
+function _countBids(bid) {
+  const bidCount = bid.sizes
```
Comment (dugwood) [inline src/adapters/indexExchange.js]: Why do you have this unused variable?

Comment (protonate) [inline src/adapters/indexExchange.js]: cruft, will remove.

File: src/bidmanager.js [hunk 2]
```diff
@@ -40,34 +41,41 @@ function getBidders(bid) {
   return bid.bidder;
 }
 
+function add(a, b) {
```
Comment (mkendall07) [inline src/bidmanager.js]: let's put into `utils`

File: src/bidmanager.js [hunk 3]
```diff
@@ -40,34 +41,41 @@ function getBidders(bid) {
...
+  return a + b;
+}
+
+/**
+ * Bidders handle multiple requests and multiple sizes in different ways.
+ * A Bidder Adapter can export a `countBids` function if this is anything
```
Comment (mkendall07) [inline src/bidmanager.js]: should we call this "countExpectedBids" ?

**Comment by mkendall07** [comment]
I restarted travis and it still is failing with ``` Uncaught TypeError: object is not a function
  at /home/travis/build/prebid/Prebid.js/test/spec/adapters/aardvark_spec.js:10281```

Seems random but maybe not...

**Comment by protonate** [comment]
@mkendall07 review notes addressed

File: src/prebid.js [hunk 4]
```diff
@@ -847,4 +847,12 @@ $$PREBID_GLOBAL$$.getHighestCpmBids = function (adUnitCode) {
   return getWinningBids(adUnitCode);
 };
 
+/**
+ * Returns the adaptermanager.bidderRegistry
+ * @returns {*}
+ */
+$$PREBID_GLOBAL$$.getBidderRegistry = function() {
```
Comment (mkendall07) [inline src/prebid.js]: Why is this added to the global? We shouldn't need to expose this. If we need a means to store objects let's put it into a separate file that does not have global access.

**Comment by protonate** [comment]
The current IX adapter bid count [workaround](https://github.com/prebid/Prebid.js/blob/master/src/bidmanager.js#L49), while inelegant, is working. I've put this is on hold for the time being. We should determine the best way to either support custom bid counting or conforming to a standard way of doing this.

cc: @prebid/indexexchange @indexexchange

**Comment by protonate** [comment]
Stale PR, closing, if there is further interest here please reopen.
